### ISSUE TITLE
Support for Hooks with Scripts or Listeners for RemoteDatabase

### ISSUE THREAD

**Issue by ironluca** [ArcadeData/arcadedb#1884]
### ArcadeDB Version:

`24.11.1`

### OS and JDK Version:

`<OS and JDK brand and version here. You can find it in the 2nd line of the server output>`

### Expected behavior

`
Support HOOK to execute custom database routines during create, update, delete operations.
The alternate solution with listeners is not supported with RemoteDatabase; if possible this 
should be supported. HOOK is the most flexible approach and there are places in the code 
base where, it looks like, some provision has been made (sql parser). Can HOOK be supported.
`

### Actual behavior

`<add here>`

### Steps to reproduce

`<add here>`

`<An SQL script to reproduce the problem or a JUnit test case will increase **A LOT** the chance to have a quick fix>`

**Comment by gramian** [comment]
There are some facilities for this available in Java: https://docs.arcadedb.com/#Java-Events , and also this issue maybe related https://github.com/ArcadeData/arcadedb/issues/1395 .

**Comment by ironluca** [comment]
> There are some facilities for this available in Java: https://docs.arcadedb.com/#Java-Events , and also this issue maybe related #1395 .

Yes, the Java events are available; however, they are not supported for RemoteDatabase. If SQL level HOOK is supported, it would be most flexible solution or support for events with RemoteDatabase where listeners can be registered.

**Comment by lvca** [comment]
Unfortunately this is not easy to do because it would require a websocket from client to server, in order to send the callback to the client. If you need hooks, I suggest to create a Server Plugin (just a matter of implementing an interface and register it in ArcadeDB at startup (look at Server Plugins in the docs).

**Comment by ironluca** [comment]
@lvca , thanks for pointing out the server plugin, I shall explore it
